src/com/xilinx/rapidwright/design/DesignTools.java

```diff
@@ -1832,10 +1832,10 @@ public static void copyImplementation(Design src, Design dest, boolean lockPlace
 			EDIFHierCellInst cellInst = src.getNetlist().getHierCellInstFromName(e.getKey());
 			destNetlist.migrateCellAndSubCells(cellInst.getCellType());
 			EDIFHierCellInst bbInst = destNetlist.getHierCellInstFromName(e.getValue());
-			bbInst.getInst().setCellType(cellInst.getCellType());
-            for(EDIFPortInst portInst : bbInst.getInst().getPortInsts()) {
-            	portInst.getPort().setParentCell(cellInst.getCellType());
-            }
+			bbInst.getInst().setCellTypeRaw(cellInst.getCellType());
+			for(EDIFPortInst portInst : bbInst.getInst().getPortInsts()) {
+				portInst.getPort().setParentCell(cellInst.getCellType());
+			}
```

Comment:
@clavin-xlnx Are you familiar with this block of code?
What's the reason for forcibly setting the cell type but not updating its portInsts?

Reply:
This code does update the portInsts, just independently, in lines 1836-1838.  These lines theoretically could be swapped for a call to the newly refactored `setCellType()`.

Reply:
It looks like those lines don't update the portInsts in the same way:
https://github.com/Xilinx/RapidWright/blob/b9d72631f0ee2b9fdccc2c846ccc0a98f0704f06/src/com/xilinx/rapidwright/edif/EDIFCellInst.java#L200-L210
Lines 1836-1838 update the parent cell of the portInsts' ports, but `setCellType` updates the portInsts themselves. Are those two things equivalent?

It doesn't look like we have any test coverage on `DesignTools.copyImplementation()` currently for me to be able to make this change with confidence.

Reply:
Oh wait, we do have one (admittedly specific) test that calls `DesignTools.copyImplementation()`: https://github.com/Xilinx/RapidWright/blob/131331d75ae66678901e5eb614fd5145428eb0eb/test/com/xilinx/rapidwright/design/TestDesignTools.java#L59

Reply:
Okay, I think changing the cell that the port belongs to is equivalent to setting a new port acquired from that new cell.